test/e2e/fixture-builder.js

```diff
@@ -0,0 +1,1002 @@
+        recoveryPhraseReminderLastShown: 1665507600000,
+        showPortfolioTooltip: true,
+        showTestnetMessageInDropdown: true,
+        trezorModel: null,
+        usedNetworks: {
+          '0x1': true,
+          '0x5': true,
+          '0x539': true,
+        },
+      },
+      NetworkController: {
+        network: '1337',
+        networkDetails: {
+          EIPS: {
+            1559: false,
+          },
+        },
+        previousProviderStore: {
+          ticker: 'ETH',
+          type: 'rpc',
+          rpcUrl: 'http://localhost:8545',
+          chainId: '0x539',
+          nickname: 'Localhost 8545',
+        },
+        provider: {
+          ticker: 'ETH',
+          type: 'rpc',
+          rpcUrl: 'http://localhost:8545',
+          chainId: '0x539',
+          nickname: 'Localhost 8545',
+        },
+      },
+      PreferencesController: {
+        advancedGasFee: null,
+        currentLocale: 'en',
+        customNetworkListEnabled: false,
+        dismissSeedBackUpReminder: false,
+        featureFlags: {
+          showIncomingTransactions: true,
+        },
+        forgottenPassword: false,
+        frequentRpcListDetail: [
+          {
+            chainId: '0x539',
+            nickname: 'Localhost 8545',
+            rpcPrefs: {},
+            rpcUrl: 'http://localhost:8545',
+            ticker: 'ETH',
+          },
+        ],
+        identities: {},
+        infuraBlocked: false,
+        ipfsGateway: 'dweb.link',
+        knownMethodData: {},
+        ledgerTransportType: 'webhid',
+        lostIdentities: {},
+        openSeaEnabled: false,
+        preferences: {
+          hideZeroBalanceTokens: false,
+          showFiatInTestnets: false,
+          showTestNetworks: false,
+          useNativeCurrencyAsPrimaryCurrency: true,
+        },
+        theme: 'light',
+        useBlockie: false,
+        useCollectibleDetection: false,
+        useNonceField: false,
+        usePhishDetect: true,
+        useTokenDetection: false,
+      },
+      SmartTransactionsController: {
+        smartTransactionsState: {
+          fees: {},
+          liveness: true,
+          smartTransactions: {
+            '0x1': [],
+          },
+        },
+      },
+      TokensController: {
+        allDetectedTokens: {},
+        allIgnoredTokens: {},
+        allTokens: {},
+        detectedTokens: [],
+        ignoredTokens: [],
+        suggestedAssets: [],
+        tokens: [],
+      },
+      config: {},
+      firstTimeInfo: {
+        date: 1665507600000,
+        version: '10.21.0',
+      },
+    },
+  };
+}
+
+class FixtureBuilder {
+  constructor({ onboarding = false } = {}) {
+    this.fixture = onboarding === true ? onboardingFixture() : defaultFixture();
+  }
+
+  withAddressBookController(data) {
+    Object.assign(
+      this.fixture.data.AddressBookController
+        ? this.fixture.data.AddressBookController
+        : (this.fixture.data.AddressBookController = {}),
+      data,
+    );
```

Comment:
The following controllers were removed from either/both the onboarding fixture and the default fixture, so we set it as an empty object if it does not exist:
AddressBookController, CollectiblesController, IncomingTransactionsController, PermissionLogController, ThreeBoxController, TokenListController, TransactionController